contracts/PositionManager.sol

```diff
@@ -127,12 +127,10 @@ contract PositionManager is IPositionManager, ERC721Holder {
     function removePositionId(uint256 tokenId) public override onlyWhitelisted {
         for (uint256 i = 0; i < uniswapNFTs.length; i++) {
             if (uniswapNFTs[i] == tokenId) {
-                if (uniswapNFTs.length > 1) {
+                if (i + 1 != uniswapNFTs.length) {
                     uniswapNFTs[i] = uniswapNFTs[uniswapNFTs.length - 1];
-                    uniswapNFTs.pop();
-                } else {
-                    delete uniswapNFTs;
                 }
+                uniswapNFTs.pop();
                 return;
             }
```

Comment:
We miss the fix for NC3.

Reply:
also, in setModuleData, we should make a require != 0 (instead of >0) since a) is more gas efficient, b) data not always will be an number